src/components/structures/SearchBox.js

```diff
@@ -61,35 +81,42 @@ module.exports = React.createClass({
         }
     },
 
+    _clearSearch: function() {
+        this.refs.search.value = "";
+        this.onChange();
```

Comment:
You're kinda right, for reasons that are ... non-obvious. In case you're interested, and for my own reference:

Essentially the `onChange` callback is a React fabrication. Native-DOM `<input>` elements *have* an `onchange` attribute, but it is only called [when the element loses focus](https://developer.mozilla.org/en-US/docs/Web/Events/change).  

React actually listens for the [`input`](https://developer.mozilla.org/en-US/docs/Web/Events/input) event, and calls the `onChange` callback when it sees one. Apparently the DOM does *not* fire an `input` event when the `value` is changed from js  - which is probably lucky otherwise React would have to jump through hoops to avoid getting stuck in a loop.

What we *ought* to do here is stick with the react side, and just make the change through react:
```
this.setState({ searchTerm: "" });
```

That will trigger a re-render, which will update the DOM.

See also: https://facebook.github.io/react/docs/forms.html#controlled-components.